src/pydap/parsers/dmr.py

```diff
@@ -89,22 +89,29 @@ def get_atomic_attr(element):
     Int_types = ["Int16", "Int32", "Int64", "int", "Int8"]
     uInt_types = ["uInt16", "uInt32", "uInt64", "uint", "uInt8", "Char"]
     name = element.get("name")
-    value = element.find("Value").text
-    if value is None:
-        # This could be because server is TDS.
-        # If value is None still, then data is missing
-        value = element.find("Value").get("value")
+    value = []

```

Comment:
I think this code is hard enough to follow that you should add a comment or doc-string showing the different forms that attribute values can take in DAP4. For example, is the code on line 97 addressing a case like <Value>17 54 23</Value> which is an alternative to <Value>17</Value> <Value>54</Value> <Value>23</Value>?

Anyway, beyond some explanation, this looks great. It's a shame we wound up with such a disjoint specification, but that's what you get with 'design by committee.' ;-)

Reply:
Added a docstring describing the 3 forms that attrs may be defined, and that these may be combined to defined a list of attribute values. I also refactored the code to extract values from 2 different forms at once.